HVSample/src/main/java/com/microsoft/hsg/android/hvsample/MainActivity.java

```diff
@@ -116,142 +126,79 @@ public void onConnected() {
-	        	break;
-	        case 3:
-	        	intent = new Intent(MainActivity.this, VocabActivity.class);
-	        	break;
-	        case 4:
-	        	intent = new Intent(MainActivity.this, LocalVaultActivity.class);
-	        }
-        }
-        
-        if(intent != null) {
-        	startActivity(intent);
-        }
-    }
-	
-	/*private void registerHandlers() {
 
-		// Connect button
-		Button connect = (Button) findViewById(R.id.connect);
-		connect.setOnClickListener(new View.OnClickListener() {
-            public void onClick(View view) {
-            	doConnect();
-            }
-        });
-		
-		Button weightsBtn = (Button) findViewById(R.id.weights);
-		weightsBtn.setOnClickListener(new View.OnClickListener() {
-            public void onClick(View view) {
-            	if (service.isAppConnected()) {
-            		Intent intent = new Intent(MainActivity.this, WeightActivity.class);
-                    startActivity(intent);
-            	}
-            }
-        });
-		
-		Button fileUploadBtn = (Button) findViewById(R.id.fileUpload);
-		fileUploadBtn.setOnClickListener(new View.OnClickListener() {
-            @SuppressWarnings("unchecked")
-			public void onClick(View view) {
-            	if (service.isAppConnected()) {
-            		String filename = writeFile();
-            		
-            		InputStream source;
-					try {
-						source = openFileInput(filename);
-						
-						File hvFile = new File();
-						hvFile.setName(filename);
-						
-						hvClient.start();
-	            		
-	            		hvClient.asyncRequest(hvFile.uploadAsync(((HVApp)getApplication()).getCurrentRecord(), null, source), 
-	            				new MainActivityCallback(MainActivityCallback.UpdateRecords));
-					} catch (Exception e) {
-						// TODO Auto-generated catch block
-						e.printStackTrace();
-					}
-            	}
-            }
-        });
-		
-		Button vocabTest = (Button) findViewById(R.id.vocabTest);
-		vocabTest.setOnClickListener(new View.OnClickListener() {
-            public void onClick(View view) {
-            	Intent intent  = new Intent(MainActivity.this, VocabActivity.class);
-            	startActivity(intent);
-            }
-        });
-	}*/
+	public void onListItemClick(ListView parent, View v, int position, long id) {
+		TextView item = (TextView) v;
+		Intent intent = null;
+		if (mService.isAppConnected()) {
+			switch(position) {
+			case 0:
+				intent = new Intent(MainActivity.this, WeightActivity.class);
+				break;
+			case 1:
+				intent = new Intent(MainActivity.this, FileUploadActivity.class);
+				break;
+			case 2:
+				Toast.makeText(MainActivity.this, "Not implemented yet", Toast.LENGTH_LONG).show();
+				break;
+			case 3:
+				intent = new Intent(MainActivity.this, VocabActivity.class);
+				break;
+			case 4:
+				intent = new Intent(MainActivity.this, LocalVaultActivity.class);
+				break;
+			case 5:
+				intent = new Intent(MainActivity.this, ActionPlanActivity.class);
+			}
```

Comment:
It is a good habit to include break statements in all cases. Please add one here. What happens if there is a value over 5? Should we log that and throw an exception?